File: signature/wordpress.go
package signature

var WordpressSignature = Signature{
	Name:        "WordPress",
	Description: "WordPress is a free and open-source content management system (CMS) written in PHP and paired with a MySQL or MariaDB database. Features include a plugin architecture and a template system, referred to within WordPress as Themes.",
	Cpe:         "cpe:/a:wordpress:wordpress",

	DetectPattern: DetectPattern{
		Bodies: []string{
			"WordPress",
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`WordPress (\d+\.\d+\.\d+)`,
		},
	},

	OnlySameHost: true,
}

File: src/signatures/technologies/wordpress.ts
import type { Signature } from "../_types.js";
import { mysqlSignature } from "./mysql.js";
import { phpSignature } from "./php.js";

export const wordpressSignature: Signature = {
  name: "WordPress",
  description:
    "WordPress is a free and open-source content management system written in PHP and paired with a MySQL or MariaDB database. Features include a plugin architecture and a template system.",
  cpe: "cpe:/a:wordpress:wordpress",
  rule: {
    confidence: "high",
    headers: {
      "x-pingback": ".+xmlrpc\\.php$",
      link: "https://api\\.w\\.org",
      "X-Redirect-By": "WordPress",
    },
    cookies: {},
    bodies: [
      "wp-content",
      "wp-includes",
      "wp-json",
      "wp\\.com",
      "WoredPress.*(\\d+\\.\\d+(\\.\\d+)?)?",
      "shareaholic:wp_version",
      "wp-embed\\.min\\.js",
    ],
  },
  impliedSoftwares: [phpSignature.name, mysqlSignature.name],
};
